43 list post  by category

mleelaw opened the pull request:
## Description
Closes #43 

Added UI logic for filtering posts by category on explore page.
## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation update
- [ ] Other (please describe)
## Checklist:
- [ ] I have read the contributing guidelines.
- [ ] I have tested the changes.
- [ ] I have updated the documentation (if applicable).
- [ ] I have added relevant tests (if applicable).
- [ ] I have checked for any breaking changes.
## Related Issue(s)
Make sure to test after pulling down the backend branch 43 as well to run on server side

herringvoices commented on src/components/posts/PostList.jsx:
```diff
@@ -5,22 +5,50 @@ import {
+    if (categoryId === "") {
+      getAllPosts().then(setPosts);
+    } else {
+      getPostByCategoryId(categoryId).then(setPosts);
```
This is set up perfectly to handle the version of GetAllPosts I mentioned in the back-end.
Instead of having different methods, however, you would optionally pass in the categoryId as query parameters.

mleelaw commented:
@herringvoices Can you take a look at the updates I made (in this case to the postManager (removed getByCategoryId and added query param check to getAll). Adjusted PostList.jsx accordingly.

herringvoices commented:
Love it!
